controllers/reconcile_configmap.go

```diff
@@ -0,0 +1,144 @@
+	mcoKc := &mcov1.KubeletConfig{}
+	if err := cmh.cli.Get(ctx, kcKey, mcoKc); err != nil {
+		return nil, err
+	}
+
+	kubeletConfig, err := mcoKubeletConfToKubeletConf(mcoKc)
+	if err != nil {
+		return nil, err
+	}
+
+	mcps, err := getNodeGroupsMCPs(ctx, cmh.cli, cmh.instance.Spec.NodeGroups)
+	if err != nil {
+		return nil, err
+	}
+
+	mcp, err := findMCPForKubeletConfig(mcps, mcoKc)
+	if err != nil {
+		return nil, err
+	}
+
+	generatedName := rtestate.GetComponentName(cmh.instance.Name, mcp.Name)
+	return cmh.syncConfigMap(ctx, kubeletConfig, generatedName)
+}
+
+func (cmh *configMapHelper) syncConfigMap(ctx context.Context, kubeletConfig *kubeletconfigv1beta1.KubeletConfiguration, name string) (*corev1.ConfigMap, error) {
+	rendered, err := renderRTEConfig(cmh.namespace, name, kubeletConfig)
+	if err != nil {
+		klog.ErrorS(err, "rendering config", "namespace", cmh.namespace, "name", name)
+		return nil, err
+	}
+
+	cfgManifests := cfgstate.Manifests{
+		Config: rendered,
+	}
+	existing := cfgstate.FromClient(context.TODO(), cmh.cli, cmh.namespace, name)
+	for _, objState := range existing.State(cfgManifests) {
+		if err := controllerutil.SetControllerReference(cmh.instance, objState.Desired, cmh.scm); err != nil {
+			return nil, errors.Wrapf(err, "Failed to set controller reference to %s %s", objState.Desired.GetNamespace(), objState.Desired.GetName())
+		}
+		if _, err := apply.ApplyObject(context.TODO(), cmh.cli, objState); err != nil {
+			return nil, errors.Wrapf(err, "could not create %s", objState.Desired.GetObjectKind().GroupVersionKind().String())
+		}
+	}
+	return rendered, nil
+}
+
+func findMCPForKubeletConfig(mcps []*machineconfigv1.MachineConfigPool, mcoKc *mcov1.KubeletConfig) (*machineconfigv1.MachineConfigPool, error) {
+	for _, mcp := range mcps {
+		if mcoKc.Spec.MachineConfigPoolSelector == nil {
+			continue
+		}
+
+		selector, err := metav1.LabelSelectorAsSelector(mcoKc.Spec.MachineConfigPoolSelector)
+		if err != nil {
+			return nil, err
+		}
+
+		if selector.Matches(labels.Set(mcp.Labels)) {
+			return mcp, nil
+		}
+	}
+	return nil, fmt.Errorf("no match")
+}
+
+func mcoKubeletConfToKubeletConf(mcoKc *mcov1.KubeletConfig) (*kubeletconfigv1beta1.KubeletConfiguration, error) {
+	kc := &kubeletconfigv1beta1.KubeletConfiguration{}
+	err := json.Unmarshal(mcoKc.Spec.KubeletConfig.Raw, kc)
+	return kc, err
+}
+
+func renderRTEConfig(namespace, name string, klConfig *kubeletconfigv1beta1.KubeletConfiguration) (*corev1.ConfigMap, error) {
+	conf := rteconfig.Config{
+		Resources: sysinfo.Config{
+			ReservedCPUs: klConfig.ReservedSystemCPUs,
+		},
+		TopologyManagerPolicy: klConfig.TopologyManagerPolicy,
+		TopologyManagerScope:  klConfig.TopologyManagerScope,
+	}
+	data, err := yaml.Marshal(conf)
+	if err != nil {
+		return nil, err
+	}
+	return rtemanifests.CreateConfigMap(namespace, name, string(data)), nil
```

Comment:
Do we need to handle also deletion of config maps ?

Reply:
handled by the owner reference (or it's a bug)